### DISCUSSION TITLE
New tutorial: Using SLAM for Map Building!

### DISCUSSION THREAD

**Discussion by mcdoerr** [prl-mushr/mushr#14] [General]
Hi everyone!

We just introduced a new tutorial,  [Using SLAM for Map Building](https://mushr.io/tutorials/mapping/). This tutorial shows how to use [gmapping](http://wiki.ros.org/gmapping) and your car's laser scanner to create custom maps of your environment that your car can use for autonomous navigation and more. Let us know if you have any feedback or questions!

**Comment by droemer7** [comment]
Thanks for the tutorial!

I have been working on my own localization node and have just gotten to the point where I am running real world testing. I ran gmapping recently with the provided launch file - no reconfiguring - but I get a very sparse map, to the point that I didn't even bother trying to connect the dots in an image editor. I noticed in this tutorial you recommended stopping once in a while to allow it to process data, but that didn't seem to help. Making multiple passes through the same room also did not help.

What did help was lower the occupancy threshold from 0.25 (default) to 0.1. Is that something you have had to do, or are there other parameters to tweak first? It is possible that my environment is challenging for the laser - I'm currently testing in my apartment (hard floor fyi) where there are many different types of surfaces.

**Reply by mcdoerr** [reply]
Hi, glad you tried the tutorial! 

I haven't tried lowering the occupancy threshold, but that seems like a good solution. I don't believe we modified any of the ROS defaults outside of the package, either.

We also tested on hardwood floors. The additional slip may introduce some errors. What really made a difference for me (and I probably should have been more explicit in the tutorial) was not driving very far between stops. I drove about 3 feet, paused, and continued. I think it would also be possible to set your drive speed very low, and continuously drive.

We did edit the maps pretty heavily to get them to have continuous lines. We included a picture of the unedited and edited files on the tutorial -- not sure how yours compares, but it does need significant cleanup usually. 

Thanks for the tip about lowering the occupancy threshold. I'll test that out on the next run!

**Reply by droemer7** [reply]
Ok, so it sounds like my results are inline with expectation. That's fine - it only took about 30 min to touch it up and get a clean map. Editing is always going to be required.

I forgot to mention I am running with 0.5 m/s max speed set in the joy_teleop.yaml. I had to do this because I have tight spaces here, but it does also do a good job limiting slip and making it easier on the SLAM scan matcher. That could be why I didnt notice much benefit to stopping often (but this is a good practice anyway - no sense rushing the mapping process).
